utils/src/networking.rs

```diff
@@ -17,34 +17,62 @@ use uuid::Uuid;
 pub struct PrefixBucket(u64);
 
 impl PrefixBucket {
+    #[inline]
+    /// get ip prefix encoded as u64 value
     pub fn as_u64(&self) -> u64 {
         self.0
     }
 }
 
-impl From<&IpAddress> for PrefixBucket {
-    fn from(ip_address: &IpAddress) -> Self {
-        match ip_address.0 {
-            IpAddr::V4(ipv4) => {
-                let prefix_bytes = ipv4.octets();
-                Self(u64::from_be_bytes([0u8, 0u8, 0u8, 0u8, 0u8, 0u8, prefix_bytes[0], prefix_bytes[1]]))
-            }
+impl From<Ipv4Addr> for PrefixBucket {
+    #[inline]
```

Comment:
I see `#[inline]` is added to a lot of the functions in this PR. Is this a recommended practice?

Reply:
in this case, I think it has its merits, and is good practice, probably the compiler will be smart enough to inline by itself, but if the function call is short, and we can expect it to be used a lot, it can have performance benefits by copying the function's code in directly instead of needing to jump to the call site. if you delve deep into core codebases with From implementations, that don't involve a lot of processing, you will see this used a lot. 

you can read up on the rust cook-book, regarding this, here: https://nnethercote.github.io/perf-book/inlining.html. but if you really want to, i can add benches?